### DISCUSSION TITLE
Flux Schnell nf4

### DISCUSSION THREAD

**Discussion by DenOfEquity** [lllyasviel/stable-diffusion-webui-forge#1003] [General]
***silveroxides*** has converted Flux-Schnell to nf4: https://huggingface.co/silveroxides/flux1-nf4-weights/blob/main/flux1-schnell-bnb-nf4.safetensors

edit: lllyasviel has updated already, so there is no need to manually copy config files anymore

~~While it's downloading:~~
~~1. open up the Forge install directory and go to `webui\backend\huggingface\black-forest-labs` 
2. make a directory `FLUX.1-schnell`
3. copy everything from the `FLUX.1-dev` directory into the new directory.
The **dev** configs might not be 100% correct for **schnell** (names/paths reference dev), but it works.~~

**Comment by rabidcopy** [comment]
Can confirm copying over files to a Schnell folder allows me to load and use Schnell based checkpoints. Think the only differences between the two configs is Dev has `  "_name_or_path": "../checkpoints/flux-dev/transformer",` in transformer/config.json while the Schnell config lacks that. Then Schnell's scheduler_config.json sets `  "use_dynamic_shifting": false` instead of Dev setting it to `true`. (There may be more differences but I think the text encoder configs were identical.) Edit: For clarity you want to use the Dev folder's files 1:1. Schnell's config files available from its huggingface repo will not work as Forge expects specifically Dev's config.

Was able to load and inference this block merge of Schnell and Dev with the copied folder where previously it would fail as it expected a Schnell repo folder that didn't exist. https://huggingface.co/drbaph/FLUX.1-schnell-dev-merged-fp8-4step/blob/main/FLUX.1-schnell-dev-merged-fp8-4step.safetensors

**Comment by zeittresor** [comment]
I have also played around with but cant get it to generate with the nf4 schnell model .. \backend\huggingface\black-forest-labs\FLUX.1-dev\transformer\config.json (even as a dup in ..\FLUX.1-schnell\..) might be incorrect like rabidcopy already said. I think thats correct. What ever i set in the config file its currently not generating anything for me right now but with the dev model it works.. with the schnell model the TypeError: 'NoneType' object is not iterable show up.

**Reply by rabidcopy** [reply]
> I have also played around with but cant get it to generate with the nf4 schnell model .. \backend\huggingface\black-forest-labs\FLUX.1-dev\transformer\config.json (even as a dup in ..\FLUX.1-schnell..) might be incorrect like rabidcopy already said. I think thats correct. What ever i set in the config file its currently not generating anything for me right now but with the dev model it works.. with the schnell model the TypeError: 'NoneType' object is not iterable show up.

Try just copying the Dev folder over and renaming `dev` to `schnell`, just to be clear, you don't want the files inside to be different at all. Just the files directly from Dev's folder because those are what are needed to work. I tested the dynamic shifting value and changing it to  what Schnell uses had no effect.

![image](https://github.com/user-attachments/assets/4ad1f19c-b068-4222-8ab1-f0d2096d4be4)
What you want the Schnell folder to be named. Files inside should just be 1:1 copied from Dev's folder.

**Reply by zeittresor** [reply]
thanks for your answer but i did exactly this but the result was as i already wrote. I only tried to adjust the config afterwards after it didn't work with just the copy of the flux.1-dev folder as flux.1-schnell

**Reply by zeittresor** [reply]
..but might be that it is another bug not caused of the flux experiments.. the "TypeError: 'NoneType' object is not iterable" error could also be because of a problem with xformers (like in the "Update Aug 11" Note by lllyasviel) is written, i have not tried to uninstall xformers because all other models are working fine..

**Comment by Kroke82** [comment]
The 4-step one works for me, so I'm realizing that the models are censored with the NSFW

**Comment by agustincaniglia** [comment]
I am trying to run FLUX, but I don't know where to put the models. Can someone help me?

**Reply by DenOfEquity** [reply]
standard models directory: `{forge install location}\webui\models\Stable-diffusion`.
You can make subdirectories for organisation if you want.
If Forge is already running, refresh the **Checkpoint** dropdown menu by using the 🔄button directly to the right.

**Reply by kohweiteck** [reply]
> I am trying to run FLUX, but I don't know where to put the models. Can someone help me?

Dunno about Forge but ComfyUI need to put the checkpoints models under unet 
Source:  https://en.wikibooks.org/wiki/AI_Art_Generation_Handbook/FLUX

**Comment by SillySilk** [comment]
I'm very interested in this answer myself.

**Comment by SillySilk** [comment]
Thank you! I'm at work with a laptop that only has 6 gigs of it VRAM So
I'll have to try it when I get home! Are all the pieces together enough for
to actually work yet?
Dave

**Comment by AFMSB** [comment]
Has anyone managed to finetune it, i wanted to generate images of myself with it
